src/gpio.rs

```diff
@@ -1374,6 +1374,8 @@ macro_rules! pin {
     };
 }
 
+#[cfg(feature = "async")]
```

Comment:
As per the request from @MabezDev (a) let's fold `wait_asynch.rs` into `gpio.rs` and (b) let's rename it to simply `asynch`.

Reply:
Will do